### PR TITLE
Simplify parameterset reading

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

Simplifies reading of parameterset to make it able to handle string input and also gives root a name rather than being accessed via an empty tuple. This PR also includes a wider update of the docs (cosmetic rather than content).

### CODE REVIEW

**Comment by swillner** [comment]
@znicholls let’s also talk about this in person. I would prefer adding a function ‘str_to_list’ instead of altering the API.

**Comment by znicholls** [comment]
> I would prefer adding a function ‘str_to_list’ instead of altering the API.

`str_to_list` is a function (well, a decorator but near enough)?

**Comment by znicholls** [comment]
After discussion with @swillner @rgieseke:

- [ ]  move decorator to a function
- [ ]  function throws a warning the first time it has to do type conversion
- [ ]  just fixes it in future

**Comment by znicholls** [comment]
@swillner @rgieseke this is ready to go, it should follow #92 to avoid confusion about the effect of `make black`

File: openscm/utils.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    wrapped.calls = 0
+
+    return wrapped
+
+
+@counted
+def ensure_input_is_tuple(inp):
+    if isinstance(inp, str):
+        if ensure_input_is_tuple.calls == 1:
```
Comment (swillner) [inline openscm/utils.py]: Do we really need a decorator here (that itself uses a decorated function)? The more straight forward way would be:
```suggestion
        if ensure_input_is_tuple.get_attr('calls', 0) == 1:
            ensure_input_is_tuple.calls = 1
```

**Comment by swillner** [comment]
@znicholls Do you give me a ping once you are done here and I will alter the in-code documentation, ok?

**Comment by znicholls** [comment]
> Do you give me a ping once you are done here and I will alter the in-code documentation, ok?

I'm done. My preference would be that you make a PR into this branch from a new branch (perhaps called `simplify-parameterset-reading` or from your own fork).

**Comment by swillner** [comment]
> > Do you give me a ping once you are done here and I will alter the in-code documentation, ok?
> 
> I'm done. My preference would be that you make a PR into this branch from a new branch (perhaps called `simplify-parameterset-reading` or from your own fork). Then it's easier for me to see the changes (to learn how to do the docs better next time) and I don't have to worry about force pushing in future as it won't affect your work.

cool, will do!

**Comment by znicholls** [comment]
@swillner so we leave this now until a second reviewer has checked it?

**Comment by swillner** [comment]
> @swillner so we leave this now until a second reviewer has checked it?

up to you, I relaxed the 2-person requirement and @rgieseke has already seen it when we talked about it in the last meeting...

**Comment by znicholls** [comment]
Decided

**Pull request merged**